src/it/resources/com/google/checkstyle/test/chapter4formatting/rule411optionalbracesusage/InputFormattedUseOfOptionalBraces.java

```diff
@@ -0,0 +1,293 @@
+package com.google.checkstyle.test.chapter4formatting.rule411optionalbracesusage;
+
+class InputFormattedUseOfOptionalBraces {
+  /**
+   * some javadoc..
+   *
+   * @return helper func *
+   */
+  boolean condition() {
+    return false;
+  }
+
+  /** Test do/while loops. */
+  void testDoWhile() {
+    // Valid
+    do {
+      testDoWhile();
+    } while (condition());
+
+    // Invalid
+    do testDoWhile(); // violation ''do' construct must use '{}'s.'
+    while (condition());
```

Comment:
Is this formatter or checkstyle problem?
It might be that we misinterprets guide

Reply:
the missing `{}` is their problem and if you're talking about indentation of `while` then it is correct, `while` isn't child of `do` so the indentation should not be increased for it

Reply:
good, thanks a lot.